### PR TITLE
Feat: Compliance report internal workflow for IDIR & BCeID users

### PR DESCRIPTION
**Pull request by prv-proton**
- #1003 
- #1004 
- #1005 
- #1006 
- compliance reports list view filter fix

### CODE REVIEW

File: backend/lcfs/web/api/compliance_report/update_service.py [hunk 1]
```diff
@@ -71,11 +92,21 @@ async def handle_status_change(
...
+            user, ["SUPPLIER"]
+        )
+        if not has_director_role:
+            raise HTTPException(status_code=403, detail="Forbidden.")
 
     async def handle_submitted_status(self, report: ComplianceReport):
         """Handle actions when a report is submitted."""
-
+        user = self.request.user
+        has_director_role = user_has_roles(
```
Comment (AlexZorkin) [inline backend/lcfs/web/api/compliance_report/update_service.py:105]: On a fresh database this check was failing and giving a 403. Did something change in the role logic where we're testing stings against enums?

File: backend/lcfs/web/api/compliance_report/update_service.py [hunk 2]
```diff
@@ -138,34 +169,57 @@ async def handle_submitted_status(self, report: ComplianceReport):
...
+        await self.repo.update_compliance_report(report)
 
         return calculated_summary
 
     async def handle_recommended_by_analyst_status(self, report: ComplianceReport):
         """Handle actions when a report is Recommended by analyst."""
         # Implement logic for Recommended by analyst status
-        pass
+        user = self.request.user
+        has_director_role = user_has_roles(
```
Comment (AlexZorkin) [inline backend/lcfs/web/api/compliance_report/update_service.py:196]: Please update these role checks with a variable like "has_roles" instead so we don't confuse the director value.

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
I wasn't able to successfully test the director approval due to the 403 errors. It may be related to a different merge? Can you please take a look. Code otherwise looks good.

**Comment by AlexZorkin** [comment]
I think i see what happened with the auth. we need to switch over to using enums like this
```python
has_signing_role = user_has_roles(
    user, [RoleEnum.SUPPLIER, RoleEnum.SIGNING_AUTHORITY]
)

File: backend/lcfs/web/api/compliance_report/update_service.py [hunk 3]
```diff
@@ -71,11 +92,21 @@ async def handle_status_change(
     async def handle_draft_status(self, report: ComplianceReport):
         """Handle actions when a report is set to Draft status."""
         # Implement logic for Draft status
-        pass
+        user = self.request.user
+        has_director_role = user_has_roles(
```
[the remark this answers is not in the document]
Comment (prv-proton) [reply backend/lcfs/web/api/compliance_report/update_service.py]: updated

File: backend/lcfs/web/api/compliance_report/update_service.py (hunk 1 above)
Comment (prv-proton) [reply backend/lcfs/web/api/compliance_report/update_service.py]: updated

File: backend/lcfs/web/api/compliance_report/update_service.py (hunk 2 above)
Comment (prv-proton) [reply backend/lcfs/web/api/compliance_report/update_service.py]: updated

**Comment by prv-proton** [comment]
resolved all the conflicts and comments.

**Comment by prv-proton** [comment]
> I wasn't able to successfully test the director approval due to the 403 errors. It may be related to a different merge? Can you please take a look. Code otherwise looks good.

Not sure, but I was able to complete the workflow end-to-end
image
image

**Review by AlexZorkin — APPROVED** [review]
Well done, this is a major feature completed. 👍

**Pull request merged**